Topics/Development_Process/Git/Git.md

```diff
@@ -33,4 +33,68 @@
 
         `git push origin master --force`
 
-        #Note: Use with precaution because we are using the force push. So try to limit the usage of this to directly push codes to the master branch. A great alternative would be using --force-with-lease because it does not overwrite any work on the master branch if more commits are added to the master branch. (In depth comparisons of force vs force-with-lease https://itnext.io/git-force-vs-force-with-lease-9d0e753e8c41)
\ No newline at end of file
+        #Note: Use with precaution because we are using the force push. So try to limit the usage of this to directly push codes to the master branch. A great alternative would be using --force-with-lease because it does not overwrite any work on the master branch if more commits are added to the master branch. (In depth comparisons of force vs force-with-lease https://itnext.io/git-force-vs-force-with-lease-9d0e753e8c41)
+
+5. Fetch-requests
+   The git fetch command downloads commits, files, and refs from a remote repository into your local repo.
+
+   Compared with git pull and git Fetch:
+   same point is that available to accomplish the task.
+   different point is that git fetch is much safer than git pull.
+
+5.1: Wroks with remote branches
+
+To gain a deeper understanding of how git fetch operates, it's helpful to explore the way Git manages and archives commits. Within a Git repository, there is a hidden directory named `.git/objects`. This directory serves as the storage for all commits, encompassing both those from local and remote sources. To differentiate between commits from local and remote branches, Git utilizes branch references (refs). The refs associated with local branches can be found in the `.git/refs/heads/` directory. When you run the git branch command, it displays a list of these local branch refs.
+
+
+Examples in two ways:
+1. way one:
+   ```
+   git branch
+   main
+   CSC207
+   CSC301
+   CSC209
+
+   ```
+
+   
+2. way two:
+   ```
+   /.git/refs/heads/
+   main
+   CSC207
+   CSC301
+   CSC209
+
+   ```
+
+    
```

Comment:
This part explains the difference between local branches and remote branches, and I believe this is important for beginners to understand.